code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,249 @@
+	current.faction |= "e_cult"
+
+/datum/antagonist/heretic/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, removing = FALSE)
+	current.faction -= "e_cult"
+
+/datum/antagonist/heretic/get_admin_commands()
+	. = ..()
+	.["Equip"] = CALLBACK(src,.proc/equip_cultist)
+
+/datum/antagonist/heretic/roundend_report()
+	var/list/parts = list()
+
+	var/cultiewin = TRUE
+
+	parts += printplayer(owner)
+
+	//Removed sanity if(changeling) because we -want- a runtime to inform us that the changelings list is incorrect and needs to be fixed.
+	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
+
+	if(objectives.len)
+		var/count = 1
+		for(var/datum/objective/objective in objectives)
+			if(objective.check_completion())
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='greentext'>Success!</b></span>"
+			else
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='redtext'>Fail.</span>"
+				cultiewin = FALSE
+			count++
+
+	if(cultiewin)
+		parts += "<span class='greentext'>The heretic was successful!</span>"
+	else
+		parts += "<span class='redtext'>The heretic has failed.</span>"
+
+	parts += "<b>Knowledge Researched:</b> "
+
+	var/list/knowledge_message = list()
+	for(var/X in get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		knowledge_message += "[EK.name]"
+	parts += knowledge_message.Join(", ")
+
+	return parts.Join("<br>")
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/heretic/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge += initialized_knowledge
+	researched_knowledge[initialized_knowledge] = initialized_knowledge.name
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/heretic/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/heretic/proc/has_knowledge(datum/eldritch_knowledge/wanted)
+	if(researched_knowledge[initial(wanted.name)])
+		return TRUE
+	return FALSE
+
+/datum/antagonist/heretic/proc/get_knowledge(datum/eldritch_knowledge/wanted)
+	return researched_knowledge[initial(wanted.name)] || FALSE
+
+/datum/antagonist/heretic/proc/get_all_knowledge()
+	return researched_knowledge
+
+////////////////
+// Objectives //
+////////////////
+
+/datum/objective/stalk
+	name = "spendtime"
+	var/timer = 5 MINUTES
+
+/datum/objective/stalk/update_explanation_text()
+	timer += pick(-3 MINUTES, 3 MINUTES)
```

Comment:
If you want higher randomness you could do something like this:
```suggestion
	timer += rand(-3 MINUTES, 3 MINUTES)
```
That way you get in-between values.